libsolidity/ast/Types.cpp

```diff
@@ -1869,6 +1869,30 @@ std::unique_ptr<ReferenceType> ArrayType::copyForLocation(DataLocation _location
 	return copy;
 }
 
+BoolResult ArraySliceType::isImplicitlyConvertibleTo(Type const& _other) const
+{
+	if (m_arrayType.location() == DataLocation::CallData && m_arrayType.isDynamicallySized() && m_arrayType == _other)
```

Comment:
Convert to calldata? Where does that happen?

Reply:
``abi.decode`` implicitly converts to ``bytes calldata``, if the first argument is in calldata and is implicitly convertible to ``bytes calldata``.
And the conversion itself is just a no-op.